workflow-ui/backstack-android/src/androidTest/java/com/squareup/workflow1/ui/backstack/test/BackStackContainerTest.kt

```diff
@@ -0,0 +1,175 @@
+package com.squareup.workflow1.ui.backstack.test
+
+import android.view.View
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isCompletelyDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withTagValue
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.backstack.test.fixtures.BackStackTestActivity
+import com.squareup.workflow1.ui.backstack.test.fixtures.BackStackTestActivity.TestRendering
+import com.squareup.workflow1.ui.backstack.test.fixtures.ViewStateTestView
+import com.squareup.workflow1.ui.backstack.test.fixtures.ViewStateTestView.ViewHooks
+import com.squareup.workflow1.ui.backstack.toBackStackScreen
+import org.hamcrest.Matchers.equalTo
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(WorkflowUiExperimentalApi::class)
+@RunWith(AndroidJUnit4::class)
+class BackStackContainerTest {
```

Comment:
Thanks for this. Just to be sure you know, there was already some coverage in the tests of the Tic Tac Toe app, including for config change — in particular, for restoring state in and under backstacks in modals as well.

Reply:
That's reassuring, but when I add view tree owner support I need to test even more complex interactions, and writing tests for that with the TTT app would be difficult to do and make the sample a lot messier.

Reply:
Oh hell yeah, this is great.